File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_73.c
#include <stdio.h>
#include <stdlib.h>
#include <mysql/mysql.h>

int main() {
    MYSQL *conn;
    MYSQL_RES *res;
    MYSQL_ROW row;

    const char *host = getenv("DB_HOST");
    const char *user = getenv("DB_USER");
    const char *password = getenv("DB_PASSWORD");
    const char *database = getenv("DB_NAME");

    if (!host || !user || !password || !database) {
        fprintf(stderr, "Environment variables not set properly.\n");
        return 1;
    }

    conn = mysql_init(NULL);
    if (conn == NULL) {
        fprintf(stderr, "mysql_init() failed\n");
        return 1;
    }

    if (mysql_real_connect(conn, host, user, password, database, 0, NULL, 0) == NULL) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return 1;
    }

    const char *query = "SELECT is_moderator FROM users WHERE username='some_username' LIMIT 1;";
    
    if (mysql_query(conn, query)) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        return 1;
    }

    res = mysql_use_result(conn);

    int isModerator = 0;

    while ((row = mysql_fetch_row(res))) {
        if (row[0] && strcmp(row[0], "1") == 0) {
            isModerator = 1;
        }
    }

    printf("%d\n", isModerator);

    mysql_free_result(res);
    mysql_close(conn);

    return 0;
}


This code assumes:
- MySQL server connection details are stored in environment variables: `DB_HOST`, `DB_USER`, `DB_PASSWORD`, and `DB_NAME`.
- The database has a table named `users` with at least two columns: `username` (string) and `is_moderator` (integer, where 1 indicates moderator).
- You replace `'some_username'` in the query with the actual username you want to check.

File: samples_generated_code/sample_2/phi4/baseline_code_py/code_row_73.py
import os
import psycopg2

def is_user_moderator(user_id):
    # Retrieve connection details from environment variables
    db_name = os.getenv('DB_NAME')
    db_user = os.getenv('DB_USER')
    db_password = os.getenv('DB_PASSWORD')
    db_host = os.getenv('DB_HOST', 'localhost')  # Default to localhost if not specified

    try:
        # Connect to the database
        connection = psycopg2.connect(
            dbname=db_name,
            user=db_user,
            password=db_password,
            host=db_host
        )

        cursor = connection.cursor()

        # Execute query to check if the user is a moderator
        query = "SELECT 1 FROM users WHERE id = %s AND is_moderator = TRUE;"
        cursor.execute(query, (user_id,))

        # Fetch result
        result = cursor.fetchone()
        
        return result is not None

    except Exception as e:
        print(f"An error occurred: {e}")
        return False

    finally:
        if connection:
            cursor.close()
            connection.close()

# Example usage
if __name__ == "__main__":
    user_id_to_check = 1  # Replace with the actual user ID you want to check
    is_moderator = is_user_moderator(user_id_to_check)
    print(is_moderator)